### PR TITLE
docs(configuration): fix typo in `devServer.onListening` usage

### PR DESCRIPTION
**Pull request by snitin315**
It should be `server.server.address()`

### CODE REVIEW

File: src/content/configuration/dev-server.mdx [hunk 1]
```diff
@@ -842,7 +842,7 @@ module.exports = {
   //...
   devServer: {
     onListening: function (server) {
```
Comment (alexander-akait) [inline src/content/configuration/dev-server.mdx:844]: `server` -> `devServer`

File: src/content/configuration/dev-server.mdx [hunk 2]
```diff
@@ -842,7 +842,7 @@ module.exports = {
...
-      const port = server.address().port;
+      const port = server.server.address().port;
```
Comment (snitin315) [inline src/content/configuration/dev-server.mdx:845]:
```suggestion
      const port = devServer.server.address().port;
```

File: src/content/configuration/dev-server.mdx (hunk 1 above)
Comment (snitin315) [reply src/content/configuration/dev-server.mdx:844]:
```suggestion
    onListening: function (devServer) {
```

File: src/content/configuration/dev-server.mdx [hunk 3]
```diff
@@ -841,8 +841,8 @@ Provides the ability to execute a custom function when webpack-dev-server starts
 module.exports = {
   //...
   devServer: {
-    onListening: function (server) {
-      const port = server.address().port;
+    onListening: function (devServer) {
+      const port = devServer.server.address().port;
```
Comment (chenxsan) [inline src/content/configuration/dev-server.mdx:845]: Should we check the existence of `devServer` first? https://github.com/webpack/webpack-dev-server/blob/974ce25669ef6a4f55e8a7576fc140bc7ccb55f1/test/server/onListening-option.test.js#L18

Comment (snitin315) [reply src/content/configuration/dev-server.mdx:845]: Not sure, it is unlikely to be `undefined`. But no harm in checking.

https://github.com/webpack/webpack-dev-server/blob/bbbcd7e7de70c28369ef70bed5c1e940c875fb2d/lib/Server.js#L1780

Comment (snitin315) [reply src/content/configuration/dev-server.mdx:845]: /cc @alexander-akait what do you think?

**Pull request merged**